ckanext/s3filestore/uploader.py

```diff
@@ -128,14 +129,21 @@ def _cache_get(self, key):
             cache_value = six.ensure_text(cache_value)
         return cache_value
 
-    def _cache_put(self, key, value):
-        ''' Set a value in the cache, if enabled, with an appropriate expiry.
+    def _cache_put_url(self, key, value):
```

Comment:
unless we will be doing other things (like appending /acl or visability or all etc), lets keep it simple and let this just take a ttl value param and have the default as ```self.signed_url_cache_window```  if None is passed in, so that  the ttl can be injected when doing acl stuff.

Reply:
Well, there's already a logical split based on whether signed URL caching is enabled, and I've just added another to automatically distinguish signed from unsigned URLs.